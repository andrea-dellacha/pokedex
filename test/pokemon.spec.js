import { shallowMount } from '@vue/test-utils'
import PokemonPage from '@/pages/_pokemon.vue'

const get = jest.fn(() => Promise.resolve({}))
describe('PokemonPage', () => {
  const dispatch = jest.fn((n) => {})
  const commit = jest.fn((n) => {})

  const wrapper = shallowMount(PokemonPage, {
    mocks: {
      $store: {
        dispatch,
        commit,
        state: {
          pokemon: {
            pokedexApi: 'http',
          }
        }
      },
      $axios: {
        $get: get
      },
      $route: {
        params: {
          pokemon: 5
        }
      }
    },
    computed: {
      pokedexApi () {
        return this.$store.state.pokemon.pokedexApi
      }
    }
  })
  test('is a Vue instance', () => {
    expect(wrapper.vm).toBeTruthy()
  })

  test('renders correctly', () => {
    expect(wrapper.element).toBeDefined()
    expect(wrapper.element).toMatchSnapshot()
  })
})
