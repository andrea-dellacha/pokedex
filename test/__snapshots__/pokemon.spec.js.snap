// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPage renders correctly 1`] = `
<div
  class="pokemon__section"
>
  <div
    class="loader"
  >
    <div
      class="loader__pokeball -duck1"
    />
     
    <div
      class="loader__pokeball -duck2"
    />
     
    <div
      class="loader__pokeball"
    />
     
    <div
      class="loader__pikachu"
    />
     
    <div
      class="loader__ash"
    />
  </div>
</div>
`;
